editor.tsx

```diff
@@ -34,4 +49,126 @@ export const updateComponentProps = async (
   }
 
   return new Response(null, { status });
-};
+}
+
+export interface ComponentPreview {
+  componentLabel: string;
+  component: string;
+  link: string;
+}
+
+function mapComponentsToPreview(
+  [componentPath, componentModule]: [string, Module],
+) {
+  const { schema } = componentModule;
+
+  if (!schema) {
+    return;
+  }
+
+  const componentName = componentNameFromPath(componentPath);
+
+  return {
+    link: `/live/api/components/${componentName}`,
+    component: componentName,
+    componentLabel: schema.title ?? componentName,
+  };
+}
+
+export function componentsPreview(
+  req: Request,
+) {
+  const url = new URL(req.url);
+
+  if (!isPrivateDomain(url.hostname)) {
+    return new Response("Not found", { status: 404 });
+  }
+
+  const { start, end, printTimings } = createServerTiming();
+  const manifest = getManifest();
+
+  start("map-components");
+  const components: ComponentPreview[] = Object.entries(manifest.components)
+    .map(
+      mapComponentsToPreview,
+    ).filter(
+      (componentPreviewData): componentPreviewData is ComponentPreview =>
+        Boolean(componentPreviewData),
+    );
+
+  const islands: ComponentPreview[] = Object.entries(manifest.islands).map(
+    mapComponentsToPreview,
+  ).filter((componentPreviewData): componentPreviewData is ComponentPreview =>
+    Boolean(componentPreviewData)
+  );
```

Comment:
Out of curiosity: Why is this timed? Is there any perf bottleneck?

Reply:
We defined that all requests should have Timings. In future we will save these data